src/Context/Video-Context.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { createContext, useContext, useReducer } from 'react';
+import { reducerFunc, initialAuthState } from '../Reducer/Auth-Reducer';
+import { dispatchFunc, initialState } from '../Reducer/Video-Reducer';
+
+const VideoContext = createContext();
+
+export function VideoProvider({ children }) {
+    
+    const [state, dispatch] = useReducer(dispatchFunc,initialState)
+
+    return (
+        <VideoContext.Provider value={{ state, dispatch}}>
+            {children}
+        </VideoContext.Provider>
+    )
+}
+
+export function useVideo() {
+    return useContext(VideoContext);
+}
+
+
+// Authentication Context
+const AuthContext = createContext();
```

Comment:
Can move the code from this line onwards to separate file called `AuthProvider.js`